File: clients/java/src/main/java/com/svix/coyote/apis/AuthToken.java
// this file is @generated
package com.svix.coyote.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.coyote.ApiException;
import com.svix.coyote.HttpClient;
import com.svix.coyote.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import lombok.Getter;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.coyote.models.AuthTokenCreateIn;
import com.svix.coyote.models.AuthTokenCreateOut;
import com.svix.coyote.models.AuthTokenDeleteIn;
import com.svix.coyote.models.AuthTokenDeleteOut;
import com.svix.coyote.models.AuthTokenExpireIn;
import com.svix.coyote.models.AuthTokenExpireOut;
import com.svix.coyote.models.AuthTokenListIn;
import com.svix.coyote.models.AuthTokenRotateIn;
import com.svix.coyote.models.AuthTokenRotateOut;
import com.svix.coyote.models.AuthTokenUpdateIn;
import com.svix.coyote.models.AuthTokenUpdateOut;
import com.svix.coyote.models.AuthTokenVerifyIn;
import com.svix.coyote.models.AuthTokenVerifyOut;
import com.svix.coyote.models.ListResponseAuthTokenOut;

public class AuthToken {
    private final HttpClient client;

    public AuthToken(HttpClient client) {
        this.client = client;
    }

    /** Create Auth Token */
    public AuthTokenCreateOut create(
        final AuthTokenCreateIn authTokenCreateIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/create");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenCreateIn,
            AuthTokenCreateOut.class
        );
    }

    /** Expire Auth Token */
    public AuthTokenExpireOut expire(
        final AuthTokenExpireIn authTokenExpireIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/expire");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenExpireIn,
            AuthTokenExpireOut.class
        );
    }

    /** Delete Auth Token */
    public AuthTokenDeleteOut delete(
        final AuthTokenDeleteIn authTokenDeleteIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/delete");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenDeleteIn,
            AuthTokenDeleteOut.class
        );
    }

    /** Verify Auth Token */
    public AuthTokenVerifyOut verify(
        final AuthTokenVerifyIn authTokenVerifyIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/verify");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenVerifyIn,
            AuthTokenVerifyOut.class
        );
    }

    /** List Auth Tokens */
    public ListResponseAuthTokenOut list(
        final AuthTokenListIn authTokenListIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/list");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenListIn,
            ListResponseAuthTokenOut.class
        );
    }

    /** Update Auth Token */
    public AuthTokenUpdateOut update(
        final AuthTokenUpdateIn authTokenUpdateIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/update");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenUpdateIn,
            AuthTokenUpdateOut.class
        );
    }

    /** Rotate Auth Token */
    public AuthTokenRotateOut rotate(
        final AuthTokenRotateIn authTokenRotateIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/rotate");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenRotateIn,
            AuthTokenRotateOut.class
        );
    }
}

File: z-clients/cli/src/cmds/api/auth_token.rs
// this file is @generated
use clap::{Args, Subcommand};
use diom_client::DiomClient;

use super::AuthTokenNamespaceArgs;

#[derive(Args)]
#[command(args_conflicts_with_subcommands = true, flatten_help = true)]
pub struct AuthTokenArgs {
    #[command(subcommand)]
    pub command: AuthTokenCommands,
}

#[derive(Subcommand)]
pub enum AuthTokenCommands {
    Namespace(AuthTokenNamespaceArgs),
    /// Create Auth Token
    Create {
        auth_token_create_in: crate::json::JsonOf<diom_client::models::AuthTokenCreateIn>,
    },
    /// Expire Auth Token
    Expire {
        auth_token_expire_in: crate::json::JsonOf<diom_client::models::AuthTokenExpireIn>,
    },
    /// Delete Auth Token
    Delete {
        auth_token_delete_in: crate::json::JsonOf<diom_client::models::AuthTokenDeleteIn>,
    },
    /// Verify Auth Token
    Verify {
        auth_token_verify_in: crate::json::JsonOf<diom_client::models::AuthTokenVerifyIn>,
    },
    /// List Auth Tokens
    List {
        auth_token_list_in: crate::json::JsonOf<diom_client::models::AuthTokenListIn>,
    },
    /// Update Auth Token
    Update {
        auth_token_update_in: crate::json::JsonOf<diom_client::models::AuthTokenUpdateIn>,
    },
    /// Rotate Auth Token
    Rotate {
        auth_token_rotate_in: crate::json::JsonOf<diom_client::models::AuthTokenRotateIn>,
    },
}

impl AuthTokenCommands {
    pub async fn exec(
        self,
        client: &DiomClient,
        color_mode: colored_json::ColorMode,
    ) -> anyhow::Result<()> {
        match self {
            Self::Namespace(args) => {
                args.command.exec(client, color_mode).await?;
            }
            Self::Create {
                auth_token_create_in,
            } => {
                let resp = client
                    .auth_token()
                    .create(auth_token_create_in.into_inner())
                    .await?;
                crate::json::print_json_output(&resp, color_mode)?;
            }
            Self::Expire {
                auth_token_expire_in,
            } => {
                let resp = client
                    .auth_token()
                    .expire(auth_token_expire_in.into_inner())
                    .await?;
                crate::json::print_json_output(&resp, color_mode)?;
            }
            Self::Delete {
                auth_token_delete_in,
            } => {
                let resp = client
                    .auth_token()
                    .delete(auth_token_delete_in.into_inner())
                    .await?;
                crate::json::print_json_output(&resp, color_mode)?;
            }
            Self::Verify {
                auth_token_verify_in,
            } => {
                let resp = client
                    .auth_token()
                    .verify(auth_token_verify_in.into_inner())
                    .await?;
                crate::json::print_json_output(&resp, color_mode)?;
            }
            Self::List { auth_token_list_in } => {
                let resp = client
                    .auth_token()
                    .list(auth_token_list_in.into_inner())
                    .await?;
                crate::json::print_json_output(&resp, color_mode)?;
            }
            Self::Update {
                auth_token_update_in,
            } => {
                let resp = client
                    .auth_token()
                    .update(auth_token_update_in.into_inner())
                    .await?;
                crate::json::print_json_output(&resp, color_mode)?;
            }
            Self::Rotate {
                auth_token_rotate_in,
            } => {
                let resp = client
                    .auth_token()
                    .rotate(auth_token_rotate_in.into_inner())
                    .await?;
                crate::json::print_json_output(&resp, color_mode)?;
            }
        }

        Ok(())
    }
}
